iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,200 +38,298 @@ class Duration;
+
+    /// @brief Constructs a new Duration object from microseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as microseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration microseconds(const T us);
+    static constexpr Duration microseconds(const T value) noexcept;
+
+    /// @brief Constructs a new Duration object from milliseconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as milliseconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration milliseconds(const T ms);
+    static constexpr Duration milliseconds(const T value) noexcept;
+
+    /// @brief Constructs a new Duration object from seconds
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as seconds
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration seconds(const T seconds);
+    static constexpr Duration seconds(const T value) noexcept;
+
+    /// @brief Constructs a new Duration object from minutes
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as minutes
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration minutes(const T min);
+    static constexpr Duration minutes(const T value) noexcept;
+
+    /// @brief Constructs a new Duration object from hours
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as hours
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration hours(const T hours);
+    static constexpr Duration hours(const T value) noexcept;
+
+    /// @brief Constructs a new Duration object from days
+    /// @tparam T is an integer type for the value
+    /// @param[in] value as days
+    /// @return a new Duration object
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
     template <typename T>
-    static constexpr Duration days(const T days);
+    static constexpr Duration days(const T value) noexcept;
+
+    // END CREATION FROM STATIC FUNCTIONS
+
+    // BEGIN CONSTRUCTORS AND ASSIGNMENT
+
+    /// @brief Constructs a Duration from seconds and nanoseconds
+    /// @param[in] seconds portion of the duration
+    /// @param[in] nanoseconds portion of the duration
+    constexpr Duration(const uint64_t seconds, const uint32_t nanoseconds) noexcept;
```

Comment:
I would make this ctor private since it somehow defeats the purpose of the readability.

```cpp
// good
auto blubb = Duration::seconds(123);
// bad
auto bla = Duration(123, 123); // what is it?
```

If this is really needed I would recommend using the conversion ctor which takes `timeval` or `itimerspec`.